base/cvd/cuttlefish/host/commands/cvd/selector/instance_group_record.h

```diff
@@ -51,8 +51,12 @@ class LocalInstanceGroup {
   void SetProductOutPath(const std::string& product_out_path);
   TimeStamp StartTime() const;
   void SetStartTime(TimeStamp time);
-  const std::vector<cvd::Instance>& Instances() const { return instances_; }
-  std::vector<cvd::Instance>& Instances() { return instances_; }
+  const google::protobuf::RepeatedPtrField<cvd::Instance>& Instances() const {
+    return group_proto_.instances();
+  };
+  google::protobuf::RepeatedPtrField<cvd::Instance>& Instances() {
+    return *group_proto_.mutable_instances();
+  };
```

Comment:
In the future, do you think we could head in a direction of avoiding exposing the inner proto objects? From [go/UnhealthyIdioms#missing-abstractions-sea-of-details](https://go.corp.google.com/UnhealthyIdioms#missing-abstractions-sea-of-details):

> Systems sometimes lack key abstractions that would allow for conceptual chunking and ease of understanding. This may be seen as a sea of details, where you have to read and understand large swathes of the code and infer what it collectively means.
> - ...
> - Data Class
>   - Dumb data objects that only have getters and setters, with no behavior.
> - Unencapsulated Proto Buffers
>   - This is a special case of Data Class that is often seen in Google code.

The downside of exposing proto objects is we give up control over the API of this type, and can't introduce more behavior like validation in setters or member functions that calculate a response based on other fields.

Reply:
Agreed